### PR TITLE
python3Packages.hebg: init 0.2.4

### PR DESCRIPTION
**Pull request by MathisFederico**
My first contribution, please tell me if I did anything wrong !

## Description of changes

Add the [hebg python library](https://github.com/IRLL/HEB_graphs/)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/hebg/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "hebg";
+  version = "0.2.4";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-11bz+FbnaEVLiXT1eujMw8lvABlzVOeROOsdVgsyfjQ=";
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/hebg/default.nix:29]: - #325892
```suggestion
    hash = "sha256-11bz+FbnaEVLiXT1eujMw8lvABlzVOeROOsdVgsyfjQ=";
```

File: pkgs/development/python-modules/hebg/default.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    scipy
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-check
+    pytest-mock
+  ];
+
+  meta = with lib; {
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/hebg/default.nix:51]:
```suggestion
  meta = {
```

File: pkgs/development/python-modules/hebg/default.nix [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    description = "Hierarchial Explainations of Behavior as Graph";
+    homepage = "https://github.com/IRLL/HEB_graphs";
+    downloadPage = "https://github.com/IRLL/HEB_graphs";
+    changelog = "https://github.com/IRLL/HEB_graphs/releases";
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ automathis ];
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/hebg/default.nix:57]:
```suggestion
    license = lib.licenses.gpl3;
    maintainers = with lib.maintainers; [ automathis ];
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/hebg/default.nix (hunk 2 above)
Comment (MathisFederico) [reply pkgs/development/python-modules/hebg/default.nix]: Done

File: pkgs/development/python-modules/hebg/default.nix (hunk 1 above)
Comment (MathisFederico) [reply pkgs/development/python-modules/hebg/default.nix]: Done

File: pkgs/development/python-modules/hebg/default.nix (hunk 3 above)
Comment (MathisFederico) [reply pkgs/development/python-modules/hebg/default.nix]: Done

File: pkgs/development/python-modules/hebg/default.nix [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    pytest-check
+    pytest-mock
+  ];
+
+  meta = {
+    description = "Hierarchial Explainations of Behavior as Graph";
+    homepage = "https://github.com/IRLL/HEB_graphs";
+    downloadPage = "https://github.com/IRLL/HEB_graphs";
+    changelog = "https://github.com/IRLL/HEB_graphs/releases";
+    license = lib.licenses.gpl3;
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/hebg/default.nix:56]:
```suggestion
    license = lib.licenses.gpl3Only;
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/hebg/default.nix (hunk 4 above)
Comment (MathisFederico) [reply pkgs/development/python-modules/hebg/default.nix]: Done

**Comment by MathisFederico** [comment]
Is there something I should do now @Sigmanificient ?

**Comment by Sigmanificient** [comment]
Unfortunately, I don't have the ability to merge, so you have to wait for someone that can.

**Comment by MathisFederico** [comment]
@natsukium Sorry to bother you, but what should I do now ?

File: pkgs/development/python-modules/hebg/default.nix [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  tqdm,
+  scipy,
+
+  # Build, dev and test dependencies
+  git,
+  setuptools-scm,
+  pytestCheckHook,
+  pytest-check,
+  pytest-mock,
+  sphinx,
```
Comment (natsukium) [inline pkgs/development/python-modules/hebg/default.nix:19]:
```suggestion
```
unused

File: pkgs/development/python-modules/hebg/default.nix [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  version = "0.2.4";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-11bz+FbnaEVLiXT1eujMw8lvABlzVOeROOsdVgsyfjQ=";
+  };
+
+  build-system = [
+    git
```
Comment (natsukium) [inline pkgs/development/python-modules/hebg/default.nix:33]: Do we really need git to build?

File: pkgs/development/python-modules/hebg/default.nix [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    setuptools-scm
+  ];
+
+  propagatedBuildInputs = [
```
Comment (natsukium) [inline pkgs/development/python-modules/hebg/default.nix:37]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/hebg/default.nix [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-check
+    pytest-mock
+  ];
+
+  meta = {
+    description = "Hierarchial Explainations of Behavior as Graph";
```
Comment (natsukium) [inline pkgs/development/python-modules/hebg/default.nix:52]:
```suggestion
    description = "Hierarchial Explanations of Behavior as Graph";
```

File: pkgs/development/python-modules/hebg/default.nix [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+  ];
+
+  propagatedBuildInputs = [
+    networkx
+    matplotlib
+    numpy
+    tqdm
+    scipy
+  ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/hebg/default.nix:44]: Could you add `pythonImportsCheck` for a smoke test?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck-using-pythonimportscheck

File: pkgs/development/python-modules/hebg/default.nix [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-check
+    pytest-mock
+  ];
+
+  meta = {
+    description = "Hierarchial Explainations of Behavior as Graph";
+    homepage = "https://github.com/IRLL/HEB_graphs";
+    downloadPage = "https://github.com/IRLL/HEB_graphs";
```
Comment (natsukium) [inline pkgs/development/python-modules/hebg/default.nix:54]:
```suggestion
```
It is the same as homepage and does not need to be set.

File: pkgs/development/python-modules/hebg/default.nix [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+    changelog = "https://github.com/IRLL/HEB_graphs/releases";
```
Comment (natsukium) [inline pkgs/development/python-modules/hebg/default.nix:55]:
```suggestion
    changelog = "https://github.com/IRLL/HEB_graphs/releases/tag/${version}";
```

**Review by natsukium — DISMISSED** [review]

File: pkgs/development/python-modules/hebg/default.nix (hunk 8 above)
Comment (MathisFederico) [reply pkgs/development/python-modules/hebg/default.nix]: Fixed the title overall (took the same as the GitHub one)

File: pkgs/development/python-modules/hebg/default.nix (hunk 5 above)
Comment (MathisFederico) [reply pkgs/development/python-modules/hebg/default.nix]: Indeed fixed

File: pkgs/development/python-modules/hebg/default.nix (hunk 6 above)
Comment (MathisFederico) [reply pkgs/development/python-modules/hebg/default.nix]: I thought it was due to the setuptools-scm, but I just tested again and it works without it indeed

File: pkgs/development/python-modules/hebg/default.nix (hunk 7 above)
Comment (MathisFederico) [reply pkgs/development/python-modules/hebg/default.nix]: Done

File: pkgs/development/python-modules/hebg/default.nix (hunk 9 above)
Comment (MathisFederico) [reply pkgs/development/python-modules/hebg/default.nix:41]: I did for hebg, I'm not sure that is what you wanted tough

File: pkgs/development/python-modules/hebg/default.nix (hunk 10 above)
Comment (MathisFederico) [reply pkgs/development/python-modules/hebg/default.nix]: Removed

File: pkgs/development/python-modules/hebg/default.nix (hunk 11 above)
Comment (MathisFederico) [reply pkgs/development/python-modules/hebg/default.nix]: Done (and checked that it works)

**Comment by natsukium** [comment]
@ofborg build python312Packages.hebg

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 335566` at 2dffbdf0 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python311Packages.hebg</li>
    <li>python312Packages.hebg</li>
  </ul>
</details>

File: pkgs/development/python-modules/hebg/default.nix [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    tqdm
+    scipy
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-check
+    pytest-mock
+  ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/hebg/default.nix:45]:
```suggestion
  env = lib.optionalAttrs stdenv.hostPlatform.isDarwin { MPLBACKEND = "Agg"; };
```
I guess this environment variable fixes tests on darwin.

File: pkgs/development/python-modules/hebg/default.nix [hunk 13]
```diff
@@ -0,0 +1,55 @@
+{
+  lib,
```
Comment (natsukium) [inline pkgs/development/python-modules/hebg/default.nix:2]:
```suggestion
  lib,
  stdenv,
```
See ofborg error.

**Comment by MathisFederico** [comment]
@natsukium any more modifications I should do ?

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**